R/Belgium.R

```diff
@@ -0,0 +1,207 @@
+    supported_region_codes = list("1" = "iso_3166_2_region",
+                                  "2" = "iso_3166_2_province"),
+    #' @field common_data_urls List of named links to raw data that are common
+    #' across levels.
+    common_data_urls = list(
+      "main" = "https://epistat.sciensano.be/Data/COVID19BE_CASES_AGESEX.csv",
+      "hosp" = "https://epistat.sciensano.be/Data/COVID19BE_HOSP.csv"
+    ),
+    #' @field level_data_urls List of named links to raw data specific to
+    #' each level of regions. For Belgium, there are only additional data for
+    #' level 1 regions.
+    level_data_urls = list(
+      "1" = list(
+        "deaths" = "https://epistat.sciensano.be/Data/COVID19BE_MORT.csv"
+      )
+    ),
+    #' @field source_data_cols existing columns within the raw data
+    source_data_cols = c("cases_new", "deaths_new"),
+
+    #' @description Set up a table of region codes for clean data
+    #' @importFrom tibble tibble tribble
+    set_region_codes = function() {
+      self$codes_lookup$`1` <- tibble::tibble(
+        level_1_region_code = c("BE-BRU", "BE-VLG", "BE-WAL"),
+        level_1_region = c("Brussels", "Flanders", "Wallonia")
+      )
+      self$codes_lookup$`2` <- tibble::tribble(
+        ~level_2_region_code,    ~level_2_region, ~level_1_region_code,
+                    "BE-VAN",        "Antwerpen",      "BE-VLG",
+                    "BE-WBR",    "BrabantWallon",      "BE-WAL",
+                    "BE-WHT",          "Hainaut",      "BE-WAL",
+                    "BE-WLG",       "Li\u00e8ge",      "BE-WAL",
+                    "BE-VLI",          "Limburg",      "BE-VLG",
+                    "BE-WLX",       "Luxembourg",      "BE-WAL",
+                    "BE-WNA",            "Namur",      "BE-WAL",
+                    "BE-VOV",   "OostVlaanderen",      "BE-VLG",
+                    "BE-VBR",    "VlaamsBrabant",      "BE-VLG",
+                    "BE-VWV",   "WestVlaanderen",      "BE-VLG",
+                    "BE-BRU",         "Brussels",      "BE-BRU"
+        )
+    },
+
+    #' @description directs to either level 1 or level 2 processing based on
+    #' request.
+    #' @importFrom dplyr select mutate
+    #' @importFrom lubridate as_date ymd_hms
+    clean_common = function() {
+
+      # vroom fails to load two lines in the main data set
+      # For now, we filter out the broken lines and replace them
+      # with the following data shim
+
+      fixed_lines <- tibble::tribble(
+        ~DATE, ~PROVINCE, ~REGION, ~AGEGROUP, ~SEX, ~CASES,
+        "2020-04-22", "Limburg", "Flanders", "50-59", "F", 10,
+        "2021-02-17", "VlaamsBrabant", "Flanders", "10-19", "M", 12
+      ) %>%
+        mutate(DATE = as.Date(DATE), CASES = as.double(CASES))
+
+      self$data$raw$main_fixed <-
+        self$data$raw$main %>%
+          filter((REGION %in% self$codes_lookup[[1]]$level_1_region
+                   | is.na(REGION))) %>%
+        bind_rows(fixed_lines)
+
+      if (self$level == "1") {
```

Comment:
this dispatch happens automatically in the new `clean` method (fancy right).

Reply:
That's a bit of the new shared-methods.R I hadn't looked at closely enough.